src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,178 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        final Node<T> link = last;
+        final Node<T> node = new Node<>(link, value, null);
+        last = node;
+        if (link == null) {
+            first = node;
+        } else {
+            link.next = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            linkFirst(value);
+        } else if (index == size) {
+            add(value);
+        } else {
+            checkIndex(index);
+            Node<T> nextNode = getNode(index);
```

Comment:
```suggestion

            Node<T> nextNode = getNode(index);
```
you have checkIndex() inside getNode()